core-bundle/tests/Messenger/WebWorkerTest.php

```diff
@@ -0,0 +1,182 @@
+        $this->assertLoggerContainsMessage('Stopping worker.');
+    }
+
+    private function createCache(array $transportsWithRunningWorkers = []): CacheItemPoolInterface
+    {
+        $cache = new ArrayAdapter();
+
+        foreach ($transportsWithRunningWorkers as $transport) {
+            $item = $cache->getItem('contao-web-worker-'.$transport);
+            $item->expiresAfter(60);
+            $cache->save($item);
+        }
+
+        return $cache;
+    }
+
+    private function triggerWebWorker(): void
+    {
+        $this->eventDispatcher->dispatch(new TerminateEvent(
+            $this->createMock(HttpKernelInterface::class),
+            new Request(),
+            new Response(),
+        ));
+    }
+
+    private function triggerRealWorkers(array $transports): void
+    {
+        $listener = static function (WorkerRunningEvent $event): void {
+            if ($event->isWorkerIdle()) {
+                $event->getWorker()->stop();
+            }
+        };
+
+        $this->eventDispatcher->addListener(WorkerRunningEvent::class, $listener);
+        $input = new ArrayInput([
+            'receivers' => $transports,
+        ]);
+
+        $this->command->run($input, new NullOutput());
+        $this->eventDispatcher->removeListener(WorkerRunningEvent::class, $listener);
+    }
+
+    private function createConsumeCommand(): void
+    {
+        $receiverLocator = new Container();
+        $receiverLocator->set('transport-1', $this->createMock(ReceiverInterface::class));
+        $receiverLocator->set('transport-2', $this->createMock(ReceiverInterface::class));
+        $receiverLocator->set('transport-3', $this->createMock(ReceiverInterface::class));
+
+        $this->command = new ConsumeMessagesCommand(
+            $this->createMock(RoutableMessageBus::class),
+            $receiverLocator,
+            $this->eventDispatcher,
+            $this->logger,
+        );
+    }
+
+    private function addEventsToEventDispatcher(WebWorker $webWorker): void
+    {
+        $this->eventDispatcher->addListener(
+            WorkerStartedEvent::class,
+            static function (WorkerStartedEvent $event) use ($webWorker): void {
+                $webWorker->onWorkerStarted($event);
+            },
+        );
+        $this->eventDispatcher->addListener(
+            WorkerRunningEvent::class,
+            static function (WorkerRunningEvent $event) use ($webWorker): void {
+                $webWorker->onWorkerRunning($event);
+            },
+        );
+        $this->eventDispatcher->addListener(
+            TerminateEvent::class,
+            static function (TerminateEvent $event) use ($webWorker): void {
+                $webWorker->onKernelTerminate($event);
+            },
+        );
+    }
+
+    private function assertLoggerContainsMessage(string $message): void
+    {
+        // Make phpstan happy
+        if (!method_exists($this->logger, 'getLogs')) {
+            $this->fail('Wrong logger.');
+        }
```

Comment:
It seems wrong to add code just for PHPStan. Can we not use the new `@phpstan-ignore …` annotation instead?

Reply:
Done.